fxgl-samples/src/main/java/sandbox/InventorySample.java

```diff
@@ -0,0 +1,199 @@
+        settings.setWidth(900);
+        settings.setHeightFromRatio(16/9.0);
+    }
+
+    @Override
+    protected void initGameVars(Map<String, Object> vars) {
+        vars.put("wood", 5);
+        vars.put("stone", 3);
+        vars.put("crystal", 1);
+    }
+
+    @Override
+    protected void initGame() {
+        inventorySubScene = new InventorySubScene();
+
+        onKeyDown(KeyCode.F, "f", () -> {
+            getSceneService().pushSubScene(inventorySubScene);
+        });
+
+        pickupWood = getUIFactoryService().newButton(getip("wood").asString("Wood: %d"));
+        pickupStone = getUIFactoryService().newButton(getip("stone").asString("Stone: %d"));
+        pickupCrystal = getUIFactoryService().newButton(getip("crystal").asString("Crystal: %d"));
+        pickupWood.setStyle("-fx-text-fill: black; -fx-background-color: saddlebrown;");
+        pickupStone.setStyle("-fx-text-fill: black; -fx-background-color: grey");
+        pickupCrystal.setStyle("-fx-text-fill: black; -fx-background-color: aqua");
+
+
+        pickupWood.setOnAction(new EventHandler<ActionEvent>() {
+            @Override
+            public void handle(ActionEvent actionEvent) {
+                var inventoryItem = inventorySubScene.itemWood;
+                pickupItem(inventoryItem, "wood");
+            }
+        });
+
+        pickupStone.setOnAction(new EventHandler<ActionEvent>() {
+            @Override
+            public void handle(ActionEvent actionEvent) {
+                var inventoryItem = inventorySubScene.itemStone;
+                pickupItem(inventoryItem, "stone");
+            }
+        });
+
+        pickupCrystal.setOnAction(new EventHandler<ActionEvent>() {
+            @Override
+            public void handle(ActionEvent actionEvent) {
+                var inventoryItem = inventorySubScene.itemCrystal;
+                pickupItem(inventoryItem, "crystal");
+            }
+        });
+
+        var vbox = new VBox(5, pickupWood, pickupStone, pickupCrystal);
+
+        addUINode(vbox, 10, 10);
+
+    }
+
+    public <T> void pickupItem(ItemData<T> inventoryItem, String varKey) {
+        if (getip(varKey).get() > 0) {
+            inventorySubScene.playerInventory.add(inventoryItem);
+            inventoryItem.setQuantity(inventoryItem.getQuantity() + 1);
+            inc(inventoryItem.getName().toLowerCase(), -1);
+        }
+    }
+
+    public static void main(String[] args) {
+        launch(args);
+    }
+
+
+    public class InventorySubScene extends SubScene {
+        public InventoryView view;
+
+        public ItemData<Entity> itemWood;
+        public ItemData<Entity> itemStone;
+        public ItemData<Entity> itemCrystal;
+        public Inventory playerInventory;
+
+        public InventorySubScene() {
+
+            Entity entityWood = new Entity();
+            Entity entityStone = new Entity();
+            Entity entityCrystal = new Entity();
+
+            playerInventory = new Inventory<ItemData<Entity>>(10);

```

Comment:
The Inventory API was designed to hold target types directly, so `Inventory<Entity>`, which internally will generate the wrapper `ItemData`. Please have a look at the InventoryTest class in fxgl-trade / test to see sample usage.

Reply:
Hi Almas, 

Thank you for the feedback! I am new to this and have been trying my best so I really appreciate that.

I am having trouble creating the InventoryListView. Because of :

```
    private val items = FXCollections.observableArrayList<T>()
    private val itemsProperty = FXCollections.unmodifiableObservableList(items)

```
in Inventory.kt. I can't access the ItemData properties like name and quantity because I am working with that type T which I am making Entity or String. Any tips??

Other than that I have taken your input and am just now stuck at this part.

Thanks again,
Adam Bocco